README.md

```diff
@@ -1,81 +1,172 @@
+install [`decss-loader`][decss-loader]:
 
-CSS:
-``` css
-.Loader {
-  background: blue;
-  height: 4px;
-}
+```bash
+npm install decss-loader --save-dev
+# or
+yarn add --dev decss-loader
 ```
 
-React:
-``` js
-import { Loader } from './style.css'
+To use low-level API, or to manage the core library version, install `decss`:
 
-render () {
-  return <Loader />
-}
+```bash
+npm install decss --save
+# or
+yarn add decss
+```
+
+## Configuration
+
+```js
+// ...
+{
+  test: /\.css$/,
+  use: [
+    'style',
+    'decss/react', // 👈 Add loader (use 'decss/preact' for Preact)
+    {
+      loader: 'css',
+      options: {
+        modules: true, // 👈 You must enable modules to make it work
+        importLoaders: 1,
+        localIdentName: '[local]-[hash:base64:5]',
+      }
+    },
+    'postcss'
+  ],
+
+  // or if you prefer classic:
+
+  loader: 'style!decss/preact!css?modules&importLoaders=1&localIdentName=[local]-[hash:base64:5]&!postcss'
+},
+// ...
 ```
 
+## Usage
 
-### Boolean properties
+See [low-level API docs](https://github.com/kossnocorp/decss/blob/master/index.js) for more implementation details.
 
-CSS:
-``` css
-.Loader-isDisabled {
-  opacity: 0.5;
+### React
+
+```javascript
+import React from 'react'
+import { Button } from './style.css'
+
+function PanicButton ({ alreadyPanicking }) {
+  return (
+    <Button tag='button' color='red' disabled={alreadyPanicking}>
+      Panic
+    </Button>
+  )
 }
 ```
 
-React:
-``` js
-import { Loader } from './style.css'
+### Preact
 
-render () {
-  return <Loader isDisabled />
+```javascript
+import { h } from 'preact'
+import { Button } from './style.css'
+
+function PanicButton ({ alreadyPanicking }) {
+  return (
+    <Button tag='button' color='red' disabled={alreadyPanicking}>
+      Panic
+    </Button>
+  )
 }
 ```
 
-### Enum properties
+## Convention
 
-CSS:
-``` css
-.Loader-color-green {
-  background: green;
+### Component (`.Component`)
+
+`.Component` is a component 🤡 The name must be in class-case, e.g. `.FormInput`, `.UI`, etc:
+
+
+In this example, `.Button` represents `<Button />`.
+
+```css
+.Button {
+  color: white;
 }
 ```
 
-React:
-``` js
-import { Loader } from './style.css'
+### Boolean Prop (`.Component-propName`)
 
-render () {
-  return <Loader color='green' />
+`.*-disabled` is a boolean prop.
+
+```css
+.Button-disabled {
+  opacity: .5;
 }
 ```
 
-### Default properties
+In the example, `.Button-disabled` is applied to `.Button`
+when `disabled` prop is truthy:
+
+```javascript
+<Button disabled>
+  Whatever
+</Button>
+```
 
-CSS:
-``` css
-.Loader, 
-.Loader-color-green {
+### Enum Prop (`.Component-propName-option`)
+
+`.*-color-gray` is an enum prop.
+
+```css
+.Button-color-red {
+  background: red;
+}
+
+.Button-color-green {
   background: green;
 }
 ```
 
-React (Loder is green by default):
-``` js
-import { Loader } from './style.css'
+`.Button-color-red` is applied to `.Button` when `color` prop equals `"red"`:
```

Comment:
`.Button-color-red` is applied to the component when its `color` prop equals `"red"`: